1/3/script.sh

```diff
@@ -0,0 +1,21 @@
+#!/bin/bash
+
+echo "================ Создание динамической библиотеки ================"
+gcc -c -fPIC hello_dynamic.c -o hello_dynamic.o
+gcc -shared -o libhello_dynamic.so hello_dynamic.o
```

Comment:
`-lshared`, иначе у .so файла будет тип shared object вместо relocatable (и его можно будет запустить)

Reply:
Чтобы был тип relocatible, нужно использовать `-r` вместо `-shared`, но в этом случае ругается dlopen
```dlopen failed (1): ./libhello_shared.so: only ET_DYN and ET_EXEC can be loaded```

Reply:
It is not possible to create a dynamic library that cannot be executed by an operating system or a program loader, because a dynamic library is designed to be loaded and executed by programs at runtime.

Reply:
Да, не туда посмотрел, `lshared` некорректно
Нужно было `chmod -x ./libhello_dynamic.so` (не разрешить пользователю запускать библиотеку), но это не относится к данной теме, т.к. это из будущих лекций